src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -635,9 +635,9 @@ bool GenericConfigurationManagerImpl<ConfigClass>::IsCommissionableDeviceNameEna
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetCommissionableDeviceName(char * buf, size_t bufSize)
 {
-    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_NAME), CHIP_ERROR_BUFFER_TOO_SMALL);
-    strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_NAME);
-    return CHIP_NO_ERROR;
+    MutableCharSpan deviceNameSpan(buf, bufSize);
+
+    return GetDeviceInstanceInfoProvider()->GetDeviceName(deviceNameSpan);
```

Comment:
I don't see how this works.  Spans are not null-terminated, generally.  How is the caller of this function supposed to know where the name ends in the buffer?

Chances are, the signature of `GenericConfigurationManagerImpl<ConfigClass>::GetCommissionableDeviceName` should change to use MutableCharSpan too.

Reply:
Replaced